src/contexts/filterContext.js

```diff
@@ -0,0 +1,60 @@
+import { createContext, useContext, useReducer,useEffect } from "react";
+
+
+const filterContext=createContext()
+const FilterProvider=({children})=>{
+
+    const reducerFn=(filterState,filterAction)=>{
+        console.log(filterAction.type,"fromswitch",filterAction.payload)
+        switch(filterAction.type){
+            case "products":
+                return {...filterState,productsList:[...filterAction.payload]}
+            case "sort":
+                return {...filterState,sortBy:filterAction.payload}
+            case "vegetarian":
+                return {...filterState,vegetarian:filterAction.payload}
+            case "delivery":
+                    return {...filterState,delivery:filterAction.payload}
+            case "rating":
+                    return {...filterState,rating:filterAction.payload}
+            default:
+                return {}
+        }
+    }
+
+    const [{sortBy,productsList,vegetarian,delivery,rating},dispatchFilter]=useReducer(reducerFn,{allProducts:true,sortBy:null,productsList:[],vegetarian:true,delivery:false,rating:false})
```

Comment:
You should store the initial state value a variable and then use it here for clean code and readability purposes.